### PR TITLE
[HOPSWORKS-2690] Allow transformation functions to produce new, transformed column when creating training dataset

### PR DESCRIPTION
**Pull request by davitbzh**

### CODE REVIEW

File: python/hsfs/transformation_function.py [hunk 1]
```diff
@@ -17,17 +17,11 @@
 import inspect
 import ast
 
-from hsfs import util, engine
+from hsfs import util
 from hsfs.core import transformation_function_engine
 
 
 class TransformationFunction:
-    UDF = "@udf"
```
Comment (moritzmeister) [inline python/hsfs/transformation_function.py:25]: why are these not needed anymore?

File: python/hsfs/engine/spark.py [hunk 2]
```diff
@@ -315,14 +315,41 @@ def write_training_dataset(
...
-            dataset = dataset.withColumn(
-                feature_name, transformation_fn.transformation_fn(col(feature_name))
+            fn_registration_name = (
+                transformation_fn.name + "_" + str(transformation_fn.version)
             )
+            self._spark_session.udf.register(
+                fn_registration_name, transformation_fn.transformation_fn
+            )
+            transformation_fn_expressions.append(
+                "{fn_name:}({name:}) AS {name:}".format(
```
Comment (moritzmeister) [inline python/hsfs/engine/spark.py:332]: why do you need the `:` I think they add some formatting spaces in some cases, is that wanted?

**Review by moritzmeister — COMMENTED** [review]
general questions

File: python/hsfs/transformation_function.py (hunk 1 above)
Comment (davitbzh) [reply python/hsfs/transformation_function.py:25]: No its not necessary any more. We use `self._spark_session.udf.register` for `selectExpr`

File: python/hsfs/engine/spark.py (hunk 2 above)
Comment (davitbzh) [reply python/hsfs/engine/spark.py:332]: yeah, it doesn't make any difference. It seemed better readable for my eyes :)

**Review by moritzmeister — APPROVED** [review]

**Pull request merged**